### PR TITLE
Valkyrie changes to support keystone based auth

### PR DESCRIPTION
**Pull request by adrianjgeorge**

### CODE REVIEW

**Comment by wdschei** [comment]
The schema changes look good.

**Comment by adrianjgeorge** [comment]
I've updated the code to actually deal with bad status codes. There are a million magic number, i'll fix them monday.

File: repose-aggregator/components/filters/valkyrie-authorization-filter/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala [hunk 1]
```diff
@@ -321,27 +325,59 @@ class ValkyrieAuthorizationFilter @Inject()(configurationService: ConfigurationS
...
+      (Option(valkyrieServer.getUsername), Option(valkyrieServer.getPassword)) match {
+        case (Some(username), Some(password)) =>
+          Try(akkaServiceClient.get(cacheKey(callType, transformedTenant, contactId),
+            valkyrieServer.getUri + uri,
+            Map("X-Auth-User" -> username, "X-Auth-Token" -> password) ++ requestTracingHeader)
+          )
+        case _ =>
+          Try(akkaServiceClient.get(cacheKey(callType, transformedTenant, contactId),
+            valkyrieServer.getUri + uri,
+            Map("X-Auth-Token" -> authToken.getOrElse(null)) ++ requestTracingHeader)
```
Comment (dmnjohns) [inline repose-aggregator/components/filters/valkyrie-authorization-filter/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala]: What happens if an auth token isn't available here? How does the Akka Service Client handle a null value in a header map? I'll check the tests when I get down there.

Comment (dmnjohns) [inline repose-aggregator/components/filters/valkyrie-authorization-filter/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala]: Apparently it's legit, but I'm not sure what that looks like in the actual HTTP request. Moving right along.

Comment (adrianjgeorge) [inline repose-aggregator/components/filters/valkyrie-authorization-filter/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala]: Yeah, it works, but not looking at it has given me a better way to handle it. Update inbound.

**Comment by dmnjohns** [comment]
So. Many. Magic. Number. Replacements.

File: repose-aggregator/components/filters/valkyrie-authorization-filter/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala [hunk 2]
```diff
@@ -321,27 +325,59 @@ class ValkyrieAuthorizationFilter @Inject()(configurationService: ConfigurationS
...
+      (Option(valkyrieServer.getUsername), Option(valkyrieServer.getPassword)) match {
+        case (Some(username), Some(password)) =>
+          Try(akkaServiceClient.get(cacheKey(callType, transformedTenant, contactId),
+            valkyrieServer.getUri + uri,
+            Map("X-Auth-User" -> username, "X-Auth-Token" -> password) ++ requestTracingHeader)
+          )
+        case _ =>
+          Try(akkaServiceClient.get(cacheKey(callType, transformedTenant, contactId),
+            valkyrieServer.getUri + uri,
+            authToken.map(token => Map("X-Auth-Token" -> token)).getOrElse(Map.empty) ++ requestTracingHeader)
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/valkyrie-authorization-filter/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala]: Boom, no more ugly null.

Comment (dmnjohns) [inline repose-aggregator/components/filters/valkyrie-authorization-filter/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala]: I like it.

**Comment by dmnjohns** [comment]
This all looks good to me. Tests were passing before that last commit, so I expect them to continue passing... if time is an issue, I would be fine merging now, otherwise I would wait until tests pass again.

👍

**Pull request merged**